src/reducers/board.js

```diff
@@ -1,10 +1,27 @@
-import {ActionTypes, BoardConfigurations} from '../constants';
+import {ActionTypes, BoardConfigurations, CellStates} from '../constants';
 import {initializeBoard} from './helpers/board';
 
-const {INITIAL_STATE} = ActionTypes;
+const {INITIAL_STATE, BOARD_TILE_SELECTED} = ActionTypes;
 
 function board(state = [], action) {
   switch (action.type) {
+    case BOARD_TILE_SELECTED:
+      if (action.rack !== '' && state[action.selected.row][action.selected.col].status === CellStates.EMPTY) {
+        let newboard = state.slice();
+        newboard[action.selected.row][action.selected.col].status = CellStates.PENDING;
+        newboard[action.selected.row][action.selected.col].letter = action.rack;
```

Comment:
you're modifying this object by reference, so you're technically modifying old state.  let me know if you don't understand why that is, and we can talk about it.